### DISCUSSION TITLE
Consolidated feedback thread

### DISCUSSION THREAD

**Discussion by jcklpe** [uswds/uswds#4442] [Ideas]
I am implementing a Figma USWDS ui-kit. 

Before that at work I had to implement an Adobe XD USWDS ui-kit because the official version didn't work with the version of XD approved for our workplace. 

In the process I have come across a lot of bits and bobs, little pieces of feedback that just pile up in my brain. Rather than flood this board, I decided I will consolidate them to this thread.

I provide this feedback with the best intentions possible. Just trying to pass along my thoughts. Grains of salt and all that jazz.

**Comment by jcklpe** [comment]
Just wanted to pass up a small piece of feedback on the[ File Input component](https://designsystem.digital.gov/components/file-input/).

The entire interface is clickable, which I think is good, but the visual signifier for the action of "upload from folder" is hyperlinked text. 

Example here:
![image](https://user-images.githubusercontent.com/9451726/147603753-764845b5-197c-4443-ba46-2303d7880d1c.png)
 (note this is my version of the component that we use within our design system in which hyperlinked text is not underlined except on hover)

This all looks fine, but I think it would be better to have a button since the intended semantics of a browser is:hyperlinks are links, buttons are actions. At least this is how things are generally understood historically in terms of Web ui. 

Example of the alternative approach here:

![image](https://user-images.githubusercontent.com/9451726/147603867-6622e46c-27bf-4819-9fba-e892c838dda3.png)

I realize this kind of small thing but I think it's worth considering.

**Reply by mejiaj** [reply]
Good idea! I created an issue from this in #4498.

**Comment by jcklpe** [comment]
I didn't realize there was so many extra components. When I built my Adobe XD ui-kit at work, it didn't have all these components. A lot of these kind of seem .... weird? Not sure I get why they're separate components. 

Like Summary box? https://designsystem.digital.gov/components/summary-box/

How is that not just an info box alert? It has more information but why not just make a variation of the Alert spec that allows for longer content? 

Same thing with the Validation component: https://designsystem.digital.gov/components/validation/

or Site Alert:
https://designsystem.digital.gov/components/site-alert/

It's literally called "Alert". Ya'll already have an Alert and Slim Alert variant so why not just have a Site, and Form Validation variants?

There's the "Identifier" component. Why isn't this just part of the Footer component? 

https://designsystem.digital.gov/components/identifier/

It's not even a matter of a variant. I get that it has a separate purpose but that seems to me to be a case where there should be a Footer component with sub components.

**Comment by jcklpe** [comment]
Honestly I hate the Header component styling. I know this is mostly taste but I think it's way too dark and heavy.

**Reply by mejiaj** [reply]
We realize there are improvements that can be made to this and many other core components (Footer, Gov Banner, Identifier). But we're waiting until after USWDS 3.0 to start addressing these components. 

If we update now that could cause breaking changes to those who might not necessarily need or want these updates.

**Reply by jcklpe** [reply]
Fair!

**Comment by jcklpe** [comment]
The organization of form controls is weird. 

So there's a bunch of form control components. 

They are all treated as separate components. I do not think this is an ideal structure for the components from a design/conceptual perspective. 

A text input has a label, the actual text input box, and then error validation text. 

But this isn't all that makes up a text input in html. Where is the hint text?

But more than that, label, error validation, and hint text are not exclusive to text input. Conceivably ANY form input would require that. 

Furthermore not all such inputs are found in forms. Inputs like range slider might be found in a filter sidebar to a search interface, a text area input might be found in a chatbot interface. 

So instead within my day job contract, I define these as "input controls".

Label + hint text + input control + error validation = "Input Item Component". 

and then:

H1 Header Component + Input Item Component + Input Item Component + Input Item Component etc etc = Form Group Component. 

I think this structure is better because it's more modular. It builds up natural from atomic parts to larger and larger components.

**Comment by jcklpe** [comment]
Collection component? The collection component is defined but what about it's constituent  parts? 

Card component is defined but that doesn't appear to be what the Collection component is using. Very weird...

**Comment by jcklpe** [comment]
What's the point of the tile radio/checkbox component variants? There's no guidance on when to use one versus the other and it largely seem unnecessary to me.

**Comment by jcklpe** [comment]
Why doesn't the side navigation on the USWDS website use the USWDS side navigation component? I think it would help readability. 

Component:
![image](https://user-images.githubusercontent.com/9451726/150242293-6b1ee2e2-3886-4433-9fd3-ca8bdf7a293c.png)


What's on the website:

![image](https://user-images.githubusercontent.com/9451726/150242316-c3451c83-392d-471a-b4e3-2a5544882b57.png)


The official component is easier to read.... so why not use it?

**Comment by jcklpe** [comment]
This is related to my Input Controls feedback above but I'm not clear on why the Time Picker component needs to be a separate component from the dropdown or combo-box component:
https://designsystem.digital.gov/components/time-picker/

Seems like it should just be a variant.

**Comment by jcklpe** [comment]
Adopt Palettes!

Prose doesn't really need to be a component IMO. Really stuff like that, stuff that is immanent through out the entire application is better defined through design tokens as part of a palette. 

In our fork of USWDS we have a Type, Color, Icon, and Shape palette. I've seen this pattern used in other Design Systems so I don't think it's unique.

**Comment by jcklpe** [comment]
Gotta say, not a fan of the pagination component. I designed my own for my day job without knowing that there was an official one, but... yeah I am not very into it. 

I went with something more like this:

![image](https://user-images.githubusercontent.com/9451726/150243868-26db105f-1f13-43c7-826a-96e2e9bc7caf.png)


Realistically do people really need the option to click on the individual page link for pages 2-9,999? 

I don't think so. People only really use pagination when they know that there's a bunch of pages they need to skip. Or they click next. That's it.

**Comment by jcklpe** [comment]
What is the USWDS governance model? That would be useful to know. I have thoughts on things I could contribute but I don't know how decisions are made or if I could even provide work on those issues.

**Reply by jcklpe** [reply]
related to this: I think it would be good to have a more standardized way for people to request icons or component additions.

**Reply by mejiaj** [reply]
Everyone is welcome to create an issue or submit a pull request for improvements. With issues, we triage them and prioritize them based on roadmap goals. If it's accessibility-related or a component requires a hotfix we tend to prioritize it over enhancements.

**Reply by jcklpe** [reply]
@mejiaj pull requests are for code, and the design system isn't just code right?

As I understand and use the terms, a design system is a social constructed conceptual framework, a system of agreements on "how we do stuff".

I think separating a design system from implementation specifics is important. Component libraries are Ui kits are implementation. The design system however is more abstracted from specific implementations and can be applied across a wide range of contexts and implementations.

So a governance model is a bit more abstract and further back than what a pull request can address. How do those design positions get set in the first place? That's what I'm curious about.

**Comment by mejiaj** [comment]
Hey @jcklpe, thanks for all the feedback. Just letting you know we hear you and we'll try to respond to your questions and comments.

Overall I will say the components go through UX research, prototyping, and a lot of testing to ensure that they provide a good baseline for user needs. Anything additional we try to table as enhancements. As you can understand we have to set a limit at some point and prioritize.

If you want to get a general idea of how decisions are made you can check the [Product Roadmap](https://designsystem.digital.gov/about/product-roadmap/).

**Reply by jcklpe** [reply]
Hey that's helpful though not clear on the governance model still. It's basically just a product team under USDS though right?
